process/block/interceptedBlocks/interceptedTxBlock.go

```diff
@@ -101,11 +101,13 @@ func (inTxBody *InterceptedTxBlockBody) integrity() error {
 			return process.ErrNilTxHashes
 		}
 
-		if miniBlock.ReceiverShardID >= inTxBody.shardCoordinator.NumberOfShards() {
+		if miniBlock.ReceiverShardID >= inTxBody.shardCoordinator.NumberOfShards() &&
+			miniBlock.ReceiverShardID != sharding.MetachainShardId {
```

Comment:
This could be something more general, either in the core package or in the shard coordinator, something like isShaidIdValid

Reply:
validity it is not always the same. Not all topics are available for metachain, for example rewards.

Reply:
Yes, but this particular case actually says if a shard id is in the accepted range (smaller than the number of shards OR metachain). I figured it is useful in other situations as well. If it can't be reused now, we can leave it as is 👍